avm/ptn/app-service-lza/hosting-environment/modules/front-door/front-door.module.bicep

```diff
@@ -0,0 +1,283 @@
+import { diagnosticSettingFullType } from 'br/public:avm/utl/types/avm-common-types:0.5.1'
+
+@description('Optional. Diagnostic Settings for FrontDoor')
+param diagnosticSettings diagnosticSettingFullType[]
+
+@description('Required. Name of the AFD profile.')
+param afdName string
+
+@description('Name of the endpoint under the profile which is unique globally.')
+param endpointName string
+
+@allowed([
+  'Enabled'
+  'Disabled'
+])
+@description('AFD Endpoint State')
+param endpointEnabled string = 'Enabled'
+
+@description('Optional. Endpoint tags.')
+param tags object = {}
+
+@allowed([
+  'Standard_AzureFrontDoor'
+  'Premium_AzureFrontDoor'
+])
+@description('Required. The pricing tier (defines a CDN provider, feature list and rate) of the CDN profile.')
+param skuName string
+
+// @description('Custom Domain List')
+// param customDomains array
+
+@description('The name of the Origin Group')
+param originGroupName string
+
+@description('Origin List')
+param origins array
+
+@allowed([
+  'Block'
+  'Log'
+  'Redirect'
+])
+param wafRuleSetAction string = 'Block'
+
+@description('optional, default value Enabled. ')
+@allowed([
+  'Enabled'
+  'Disabled'
+])
+param wafPolicyState string = 'Enabled'
+
+@description('optional, default value Prevention. ')
+@allowed([
+  'Detection'
+  'Prevention'
+])
+param wafPolicyMode string = 'Prevention'
+
+// Create an Array of all Endpoint which includes customDomain Id and afdEndpoint Id
```

Comment:
Leftover - or for the future?

Reply:
For the future. It's from the original non-AVM implementation and I plan on adding support for custom domain names and compression settings at a later date.